### PR TITLE
fix: #5365, SplitButton: SplitButton popup not closing on escape button click

### PR DESCRIPTION
**Pull request by akshayaqburst**
### Defect Fixes
fix: #5365, SplitButton: SplitButton popup not closing on escape button click

### CODE REVIEW

File: components/lib/splitbutton/SplitButtonBase.js [hunk 1]
```diff
@@ -97,7 +97,8 @@ export const SplitButtonBase = ComponentBase.extend({
         onClick: null,
         onShow: null,
         onHide: null,
-        children: undefined
+        children: undefined,
+        closeOnEscape: true
```
Comment (melloware) [inline components/lib/splitbutton/SplitButtonBase.js:101]: Move this up to be in alphabetical order.

File: components/lib/splitbutton/splitbutton.d.ts [hunk 2]
```diff
@@ -239,6 +239,11 @@ export interface SplitButtonProps extends Omit<React.DetailedHTMLProps<React.HTM
      * @defaultValue false
      */
     unstyled?: boolean;
+    /**
+     * Specifies if pressing escape key should hide the Split button Popup.
+     * @defaultValue true
+     */
+    closeOnEscape?: boolean | undefined;
```
Comment (melloware) [inline components/lib/splitbutton/splitbutton.d.ts:246]: Move this up to be in alphabetical order

**Review by melloware — CHANGES_REQUESTED** [review]

File: components/lib/splitbutton/splitbutton.d.ts (hunk 2 above)
Comment (akshayaqburst) [reply components/lib/splitbutton/splitbutton.d.ts:246]: @melloware But I see other props are not sorted alphabetically! Should I sort the entire list?

**Comment by akshayaqburst** [comment]
> actually no need for `closeOnEscape` property this should be the default behavior and it is in PrimeVue with no property: https://primevue.org/splitbutton/

Okay

**Comment by akshayaqburst** [comment]
@melloware Does this look good now? Let me know if any more changes needed, thanks

File: components/lib/splitbutton/SplitButtonBase.js (hunk 1 above)
Comment (melloware) [inline components/lib/splitbutton/SplitButtonBase.js:101]: Remove this props

File: components/lib/splitbutton/SplitButton.js [hunk 3]
```diff
@@ -32,6 +33,9 @@ export const SplitButton = React.memo(
         const { ptm, cx, isUnstyled } = SplitButtonBase.setMetaData(metaData);
 
         useHandleStyle(SplitButtonBase.css.styles, isUnstyled, { name: 'splitbutton' });
+        useOnEscapeKey(overlayRef, props.closeOnEscape, (event) => {
```
Comment (melloware) [inline components/lib/splitbutton/SplitButton.js:36]: Does not have `props.closeOnEscape` it should just be on.

**Review by melloware — CHANGES_REQUESTED** [review]

File: components/lib/splitbutton/SplitButton.js (hunk 3 above)
Comment (melloware) [reply components/lib/splitbutton/SplitButton.js:36]: look at this PR for guidance: https://github.com/primefaces/primereact/pull/5361

Comment (akshayaqburst) [reply components/lib/splitbutton/SplitButton.js:36]: Oh, Okay

File: components/lib/splitbutton/SplitButtonBase.js (hunk 1 above)
Comment (akshayaqburst) [reply components/lib/splitbutton/SplitButtonBase.js]: Done

File: components/lib/splitbutton/SplitButton.js (hunk 3 above)
Comment (akshayaqburst) [reply components/lib/splitbutton/SplitButton.js]: Done, thanks.

**Review by melloware — APPROVED** [review]

**Pull request merged**